testcode/lineargraph.cpp

```diff
@@ -266,9 +266,14 @@ int main(int argc, char* argv[])
     mfem::SparseMatrix Pu;
     mfem::SparseMatrix Pp;
     mfem::SparseMatrix face_dof; // not used in this example
-    std::vector<mfem::DenseMatrix> CM_el_;
 
-    GraphCoarsen graph_coarsen(graph.GetM(), graph.GetD(), graph_topology);
+    mfem::Vector weight(graph.GetM().Size());
+    for (int i = 0; i < weight.Size(); i++)
+    {
+        weight(i) = graph.GetM()(i, i);
```

Comment:
MFEM has `graph.GetM().GetDiag(weight)` which is a bit simpler.

Reply:
Actually, I think I wanted to do `weight(i) = 1.0 / graph.GetM()(i, i)` but then I forgot to do so (I will fix that), although it makes no difference for this example in terms of results as M_ii is 1. The inversion is just to honour the notation used in `MixedMatrix`, so someone reading the code can find this consistent, and if someday the weight in this example is no longer 1. If we dont need to do the inversion, certainly `GetDiag` should be called.

Reply:
Ah ok, that makes sense why this is in a loop :+1: